contracts/KyberNetworkInterface.sol

```diff
@@ -16,5 +16,5 @@ interface KyberNetworkInterface {
         returns (uint expectedRate, uint slippageRate);
 
     function tradeWithHint(address trader, ERC20 src, uint srcAmount, ERC20 dest, address destAddress,
-        uint maxDestAmount, uint minConversionRate, address walletId, bytes hint) public payable returns(uint);
+        uint maxDestAmount, uint minConversionRate, address walletId, bytes hint) external payable returns(uint);
```

Comment:
I am not sure we can change network interface. it changes the binary of the proxy. need to discuss it.

Reply:
we have to do it to use the reentrant code.
as u suggest in your documentation...